### ISSUE TITLE
Endpoints/Delete should use empty to validate required arguments

### ISSUE THREAD

**Issue by 13leaf** [elastic/elasticsearch-php#263]
Please use empty to validate requirement.
```php
$client->delete(array("index"=>$index,"type"=>$type,"id"=>$id));
```
If `$id` is an empty string `""`,client will still performRequest` /index/type/`,which maybe a terrible mistake.(delete all type documents)
So please use `empty` instead of `isset` to validate required arguments.

**Comment by polyfractal** [comment]
`empty()` likely won't work, since it will return true for `(int) 0`, `(float) 0.0` and `(string) "0"`, which are all valid values for index/type/id.

But I agree an empty string is a Bad Thing to allow in a delete api.  We can add a check to explicitly look for an empty string.

**Comment by 13leaf** [comment]
Sorry,you are right,empty has problems! I think we can use `strlen`.

**Comment by polyfractal** [comment]
Fixed in 7b755d907113279cae6aec32fd3bb66eb997b2c5
